### PR TITLE
Sndrec32: Exchange button bitmaps with Marlett Symbols - Core 11354

### PR DESCRIPTION
**Pull request by gonzoMD**
## Purpose

_Exchange the button bitmaps with Marlett text._

JIRA issue: [CORE-11354](https://jira.reactos.org/browse/CORE-11354)

## Proposed changes

- Remove bitmap files.
- remove some now unneeded helper functions.
- Add an owner draw function for the buttons

Classic:
classic

Themed:
themed 

## TODO

Everything is addressed now :)

- [x] Find out why this only works in classic mode (May be a ROS bug. On WinXP it works in classic as well as themed mode)
- [x] Get the padding between the single symbols smaller
- [x] Color the symbols

### CODE REVIEW

File: base/applications/sndrec32/sndrec32.cpp [hunk 1]
```diff
@@ -126,19 +125,25 @@ _tWinMain(HINSTANCE hInstance,
...
-    butbmps[3] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_STOP));
-    butbmps[4] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_REC));
-
-    butbmps_dis[0] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_START_DIS));
-    butbmps_dis[1] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_END_DIS));
-    butbmps_dis[2] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_PLAY_DIS));
-    butbmps_dis[3] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_STOP_DIS));
-    butbmps_dis[4] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_REC_DIS));
+    /* Inits buttons texts */
+    buttxts[0] = TEXT("33");
```
Comment (learn-more) [inline base/applications/sndrec32/sndrec32.cpp:129]: Please add comments behind the lines what each button is.

**Comment by learn-more** [comment]
@gonzoMD Was this the kind of advice you were expecting, or do you want someone to help you more with writing this code?

**Comment by gonzoMD** [comment]
> @gonzoMD Was this the kind of advice you were expecting, or do you want someone to help you more with writing this code?

Thank you Mark. This was good. I already read about that Message, now I know that my research went into the right direction. 
Now I have to figure out how to use ownerdrawn buttons (pushbuttons seem to be a Bit tricky) and then more will follow.

File: base/system/diskpart/lang/de-DE.rc [hunk 2]
```diff
@@ -117,6 +117,14 @@ BEGIN
...
 END
 
+STRINGTABLE
+BEGIN
+    IDS_SETID_FAIL "\nDiskPart konnte den Partitionstyp nicht ändern.\n"
+    IDS_SETID_SUCCESS "\nDer Partitionstyp wurde erfolgreich geändert.\n"
+    IDS_SETID_INVALID_FORMAT "\nDas Format des Partitionstyps ist ungültig.\n"
+    IDS_SETID_INVALID_TYPE "\nDer Partitionstyp ist ungültig.\n"
+END
+
```
Comment (binarymaster) [inline base/system/diskpart/lang/de-DE.rc:127]: PR is rekt now, there are unrelated commits, never do a merge commit.

**Comment by gonzoMD** [comment]
I tried to add themes support for the owner drawn buttons like described here: https://docs.microsoft.com/de-de/windows/win32/controls/using-visual-styles
That didn't work so I checked how it's done in calc.exe and realized that everything seems to be a big hack there.
I'll leave it as is for now.

**Comment by learn-more** [comment]
> I tried to add themes support for the owner drawn buttons like described here: https://docs.microsoft.com/de-de/windows/win32/controls/using-visual-styles That didn't work so I checked how it's done in calc.exe and realized that everything seems to be a big hack there. I'll leave it as is for now.

Owner-drawn means you also have to draw the themed stuff yourself.

**Comment by gonzoMD** [comment]
> > I tried to add themes support for the owner drawn buttons like described here: https://docs.microsoft.com/de-de/windows/win32/controls/using-visual-styles That didn't work so I checked how it's done in calc.exe and realized that everything seems to be a big hack there. I'll leave it as is for now.
> 
> Owner-drawn means you also have to draw the themed stuff yourself.

According the doc, you load the Themedata into a Handle.
When the Handle is not NULL (themes are enabled) you have to use the uxtheme api to draw the stuff instead of the regular drawing api.
But the uxtheme stuff seems to be (at least partially) missing.

File: base/applications/sndrec32/sndrec32.cpp [hunk 3]
```diff
@@ -38,11 +38,10 @@ HWND main_win;
 HWND wave_win;
 HWND slider;
 HWND buttons[5];
-HBITMAP butbmps[5];
-HBITMAP butbmps_dis[5];
+LPCWSTR buttxts[5];
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:41]:
```suggestion
LPCTSTR buttxts[5];
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 4]
```diff
@@ -38,11 +38,10 @@ HWND main_win;
...
+HFONT hfButtoncaption;
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:42]:
```suggestion
HFONT hfButtonCaption;
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 5]
```diff
@@ -768,6 +770,46 @@ WndProc(HWND hWnd,
...
+                    return TRUE;
+                }
+                break;
+
+                case BUTREC_ID:
+                {
+                    DrawButton(BUTREC_ID, lParam);
+                    return TRUE;
+                }
+                break;
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:809]:
```suggestion
                case BUTSTART_ID:
                case BUTEND_ID:
                case BUTPLAY_ID:
                case BUTSTOP_ID:
                case BUTREC_ID:
                {
                    DrawButton((UINT)wParam, lParam);
                    return TRUE;
                }
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 6]
```diff
@@ -126,19 +125,25 @@ _tWinMain(HINSTANCE hInstance,
...
-    butbmps_dis[1] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_END_DIS));
-    butbmps_dis[2] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_PLAY_DIS));
-    butbmps_dis[3] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_STOP_DIS));
-    butbmps_dis[4] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_REC_DIS));
+    /* Inits buttons texts */
+    buttxts[0] = TEXT("33");    // rewind
+    buttxts[1] = TEXT("44");    // forward
+    buttxts[2] = TEXT("4");     // play
+    buttxts[3] = TEXT("g");     // stop
+    buttxts[4] = TEXT("n");     // record
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:133]:
```suggestion
    /* Init buttons texts */
    buttxts[BUTSTART_ID] = TEXT("33");
    buttxts[BUTEND_ID  ] = TEXT("44");
    buttxts[BUTPLAY_ID ] = TEXT("4");
    buttxts[BUTSTOP_ID ] = TEXT("g");
    buttxts[BUTREC_ID  ] = TEXT("n");
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 7]
```diff
@@ -880,6 +922,10 @@ WndProc(HWND hWnd,
             break;
 
         case WM_DESTROY:
+            if(hfButtoncaption)
+            {
+                DeleteObject(hfButtoncaption);
+            }
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:928]:
```suggestion
            if (hfButtonCaption)
                DeleteObject(hfButtonCaption);
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 8]
```diff
@@ -126,19 +125,25 @@ _tWinMain(HINSTANCE hInstance,
...
-    butbmps_dis[4] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_REC_DIS));
+    /* Inits buttons texts */
+    buttxts[0] = TEXT("33");    // rewind
+    buttxts[1] = TEXT("44");    // forward
+    buttxts[2] = TEXT("4");     // play
+    buttxts[3] = TEXT("g");     // stop
+    buttxts[4] = TEXT("n");     // record
+
+    /* Create a Marlett Font */
+    hfButtoncaption = CreateFont(0, 0, 0, 0,
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:136]:
```suggestion
    hfButtonCaption = CreateFont(0, 0, 0, 0,
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 9]
```diff
@@ -126,19 +125,25 @@ _tWinMain(HINSTANCE hInstance,
...
+                                 FW_BOLD,
+                                 FALSE,
+                                 FALSE,
+                                 FALSE,
+                                 SYMBOL_CHARSET,
+                                 OUT_DEFAULT_PRECIS,
+                                 CLIP_DEFAULT_PRECIS,
+                                 DEFAULT_QUALITY,
+                                 DEFAULT_PITCH|FF_DECORATIVE,
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:145]:
```suggestion
                                 DEFAULT_PITCH | FF_DECORATIVE,
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 10]
```diff
@@ -471,10 +476,10 @@ WndProc(HWND hWnd,
                     return FALSE;
                 }
 
-                /* Realize the button bmp image */
-                SendMessage(buttons[i], BM_SETIMAGE, (WPARAM)IMAGE_BITMAP, (LPARAM)butbmps[i]);
+                /* Use Marlett font on all buttons */
+                SendMessage(buttons[i], WM_SETFONT, (WPARAM)hfButtoncaption, (LPARAM)TRUE);
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:480]:
```suggestion
                SendMessage(buttons[i], WM_SETFONT, (WPARAM)hfButtonCaption, (LPARAM)TRUE);
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 11]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+    ((lpdis->rcItem.right - lpdis->rcItem.left) - size.cx + captionspacer) / 2,
+    ((lpdis->rcItem.bottom - lpdis->rcItem.top) - size.cy) / 2,
+    ETO_CLIPPED, &lpdis->rcItem, buttxts[ButtonID], _tcslen(buttxts[ButtonID]), lpdx);
+
+    if (ButtonStatus & BST_FOCUS)
+    {
+        lpdis->rcItem.left +=3;
+        lpdis->rcItem.top +=3;
+        lpdis->rcItem.right -=3;
+        lpdis->rcItem.bottom -=3;
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:990]:
```suggestion
        InflateRect(&lpdis->rcItem, -3, -3);
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 12]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+        SetTextColor(lpdis->hDC, GetSysColor(COLOR_GRAYTEXT));
+    }
+
+    DrawFrameControl(lpdis->hDC, &lpdis->rcItem, DFC_BUTTON, ButtonStyle);
+
+    GetTextExtentPoint32(lpdis->hDC, buttxts[ButtonID], _tcslen(buttxts[ButtonID]), &size);
+    ExtTextOut(lpdis->hDC,
+    ((lpdis->rcItem.right - lpdis->rcItem.left) - size.cx + captionspacer) / 2,
+    ((lpdis->rcItem.bottom - lpdis->rcItem.top) - size.cy) / 2,
+    ETO_CLIPPED, &lpdis->rcItem, buttxts[ButtonID], _tcslen(buttxts[ButtonID]), lpdx);
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:983]:
```suggestion
    ExtTextOut(lpdis->hDC,
               ((lpdis->rcItem.right - lpdis->rcItem.left) - size.cx + captionspacer) / 2,
               ((lpdis->rcItem.bottom - lpdis->rcItem.top) - size.cy) / 2,
               ETO_CLIPPED,
               &lpdis->rcItem,
               buttxts[ButtonID],
               _tcslen(buttxts[ButtonID]),
               lpdx);
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 13]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
 }
 
-void l_play_finished(void)
+void DrawButton(int ButtonID, LPARAM lParam)
+{
+    LPDRAWITEMSTRUCT lpdis = (DRAWITEMSTRUCT*)lParam;
+    SIZE size;
+    LRESULT ButtonStatus;
+    int ButtonStyle = DFCS_BUTTONPUSH;
+    int lpdx[] = {8,8}, captionspacer=0;
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:944]:
```suggestion
    int lpdx[] = {8,8}, captionspacer = 0;
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 14]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+        SetTextColor(lpdis->hDC, RGB(255, 0, 0));
+    }
+    else
+    {
+        SetTextColor(lpdis->hDC, RGB(0, 0, 0));
+    }
+
+    ButtonStatus = SendMessage(buttons[ButtonID], BM_GETSTATE, NULL, NULL);
+
+    if(ButtonStatus & BST_PUSHED)
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:963]:
```suggestion
    if (ButtonStatus & BST_PUSHED)
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 15]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+    {
+        ButtonStyle |= DFCS_PUSHED;
+    }
+    else if(ButtonStatus & BST_HOT)
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:967]:
```suggestion
    else if (ButtonStatus & BST_HOT)
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 16]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+    {
+        ButtonStyle |= DFCS_HOT;
+    }
+    else if(!IsWindowEnabled(buttons[ButtonID]))
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:971]:
```suggestion
    else if (!IsWindowEnabled(buttons[ButtonID]))
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 17]
```diff
@@ -455,8 +460,8 @@ WndProc(HWND hWnd,
             for (int i = 0; i < 5; ++i)
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:460]:
```suggestion
            for (int i = 0; i < _countof(buttons); ++i)
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 18]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+    int ButtonStyle = DFCS_BUTTONPUSH;
+    int lpdx[] = {8,8}, captionspacer=0;
+
+    if (ButtonID<BUTPLAY_ID)
+    {
+        captionspacer=8;
+    }
+
+
+    if (ButtonID >= BUTREC_ID)
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:952]:
```suggestion
    if (ButtonID < BUTPLAY_ID)
    {
        captionspacer = 8;
    }

    if (ButtonID >= BUTREC_ID)
```

File: base/applications/sndrec32/sndrec32.h [hunk 19]
```diff
@@ -105,9 +105,8 @@ LRESULT CALLBACK Buttons_proc(HWND, UINT, WPARAM, LPARAM);
 
 BOOL write_wav(TCHAR *);
 BOOL open_wav(TCHAR *);
-VOID enable_but(DWORD);
-VOID disable_but(DWORD);
 
+void DrawButton(int, LPARAM);
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.h:109]:
```suggestion
void DrawButton(UINT, LPARAM);
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 20]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
     return 0;
 }
 
-void l_play_finished(void)
+void DrawButton(int ButtonID, LPARAM lParam)
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:938]:
```suggestion
void DrawButton(UINT ButtonID, LPARAM lParam)
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 21]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+{
+    LPDRAWITEMSTRUCT lpdis = (DRAWITEMSTRUCT*)lParam;
+    SIZE size;
+    LRESULT ButtonStatus;
+    int ButtonStyle = DFCS_BUTTONPUSH;
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:943]:
```suggestion
    UINT ButtonStyle = DFCS_BUTTONPUSH;
```

File: base/applications/sndrec32/sndrec32.cpp [hunk 22]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+    }
+
+
+    if (ButtonID >= BUTREC_ID)
+    {
+        SetTextColor(lpdis->hDC, RGB(255, 0, 0));
+    }
+    else
+    {
+        SetTextColor(lpdis->hDC, RGB(0, 0, 0));
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:958]: This `DrawButton` function must restore the original color in the DC at the end.

File: base/applications/sndrec32/sndrec32.cpp [hunk 23]
```diff
@@ -889,14 +935,71 @@ WndProc(HWND hWnd,
...
+    }
+
+    ButtonStatus = SendMessage(buttons[ButtonID], BM_GETSTATE, NULL, NULL);
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:961]: What's going to happen in this `DrawButton` function if one calls it with a button ID that is >= BUTREC_ID ?

File: base/applications/sndrec32/sndrec32.cpp (hunk 7 above)
Comment (HBelusca) [reply base/applications/sndrec32/sndrec32.cpp:928]: Do we need to keep that `hfButtoncaption` around for the whole duration of the program?

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: base/applications/sndrec32/sndrec32.cpp (hunk 7 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp:928]: Not sure. The documentation states:

> When you no longer need the font, call the [DeleteObject](https://docs.microsoft.com/en-us/windows/desktop/api/wingdi/nf-wingdi-deleteobject) function to delete it.

I thought that this seems to be the correct place. When you as an experienced Win32 Dev tell me something else I'd like glad to hear your opinion. :)

File: base/applications/sndrec32/sndrec32.cpp (hunk 23 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp:961]: it sets the text color to read, for the record button. I used >= because BUTREC_ID is the bigges ID provided, but I think a == would be more correct.

File: base/applications/sndrec32/sndrec32.cpp (hunk 22 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp:958]: How exactly? It uses red only for the BUTREC_ID and instead it uses black.

Comment (HBelusca) [reply base/applications/sndrec32/sndrec32.cpp:958]: > How exactly?

By retrieving the returned value from these SetTextColor calls, and restoring it (re-calling SetTextColor with that value) at the end of the function.

File: base/applications/sndrec32/sndrec32.cpp (hunk 23 above)
Comment (HBelusca) [reply base/applications/sndrec32/sndrec32.cpp:961]: Yes, but if one calls the `DrawButton` function with a button ID that is > BUTREC_ID (greater strictly), it will throw an access violation because then you're going to access the buttons array beyond its end.

File: base/applications/sndrec32/sndrec32.cpp (hunk 7 above)
Comment (HBelusca) [reply base/applications/sndrec32/sndrec32.cpp:928]: > When you no longer need the font

Exactly, nowhere else you use that font handle, except in your helper function where you create it to do some stuff just afterwards. Thus the handle can be deleted at the end of said helper function.

File: base/applications/sndrec32/sndrec32.cpp (hunk 22 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp:958]: OK, got it. Thanks

File: base/applications/sndrec32/sndrec32.cpp (hunk 23 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp:961]: OK. I guarded the call:
```
if ((ButtonID >= BUTSTART_ID) && (ButtonID <= BUTREC_ID))
{
    ButtonStatus = SendMessage(buttons[ButtonID], BM_GETSTATE, NULL, NULL);
}
```

File: base/applications/sndrec32/sndrec32.cpp (hunk 17 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp:460]: Throws a -Werror=sign-compare

Comment (tkreuzer) [reply base/applications/sndrec32/sndrec32.cpp:460]: `for (unsigned int i = 0; i < _countof(buttons); ++i)` ?

**Comment by gonzoMD** [comment]
All done so far. Can someone, maybe @peterooch, test if this works correct with RTL languages?

**Comment by peterooch** [comment]
Before PR:
ros_snd_rec_before_marlett
After PR:
ros_snd_rec_after_marlett

The button bitmaps are being mirrored while the button text isn't.
The forward and rewind button texts should be swapped to look more logical...

File: base/applications/sndrec32/sndrec32.cpp [hunk 24]
```diff
@@ -126,19 +128,37 @@ _tWinMain(HINSTANCE hInstance,
...
-    butbmps[3] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_STOP));
-    butbmps[4] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_REC));
-
-    butbmps_dis[0] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_START_DIS));
-    butbmps_dis[1] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_END_DIS));
-    butbmps_dis[2] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_PLAY_DIS));
-    butbmps_dis[3] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_STOP_DIS));
-    butbmps_dis[4] = LoadBitmap(hInstance, MAKEINTRESOURCE(IDB_BITMAP2_REC_DIS));
+    /* Init button texts, exchange rewind and forward in case we use an RTL layout*/
+    if(layout==LAYOUT_RTL)
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:136]:
```suggestion
    if (layout == LAYOUT_RTL)
```

**Comment by gonzoMD** [comment]
> Before PR: ros_snd_rec_before_marlett After PR: ros_snd_rec_after_marlett
> 
> The button bitmaps are being mirrored while the button text isn't. The forward and rewind button texts should be swapped to look more logical...

done.

File: base/applications/sndrec32/sndrec32.cpp (hunk 7 above)
Comment (gonzoMD) [reply base/applications/sndrec32/sndrec32.cpp]: OK. I tried to reduce the usage of this HFONT by making it a variable in the scope of WndProc() instead of a global one but deleting the Object direct after using it breaks font drawing.

File: base/applications/sndrec32/CMakeLists.txt [hunk 25]
```diff
@@ -15,6 +15,6 @@ add_rc_deps(rsrc.rc ${sndrec32_rc_deps})
 add_executable(sndrec32 ${SOURCE} rsrc.rc)
 target_link_libraries(sndrec32 cpprt)
 set_module_type(sndrec32 win32gui UNICODE)
-add_importlibs(sndrec32 winmm user32 msacm32 comctl32 comdlg32 gdi32 shell32 msvcrt kernel32)
+add_importlibs(sndrec32 winmm user32 msacm32 comctl32 comdlg32 gdi32 shell32 msvcrt kernel32 uxtheme)
```
Comment (HBelusca) [inline base/applications/sndrec32/CMakeLists.txt:18]: uxtheme has to be loaded dynamically.

File: base/applications/sndrec32/sndrec32.cpp [hunk 26]
```diff
@@ -768,6 +727,39 @@ WndProc(HWND hWnd,
...
+            switch ((UINT)wParam)
+            {
+                case BUTSTART_ID:
+                case BUTEND_ID:
+                case BUTPLAY_ID:
+                case BUTSTOP_ID:
+                case BUTREC_ID:
+                {
+                    CloseThemeData (buttontheme[(UINT)wParam]);
+                    buttontheme[(UINT)wParam] = OpenThemeData (hWnd, L"Button");
```
Comment (HBelusca) [inline base/applications/sndrec32/sndrec32.cpp:740]: Please remove extra space between function name and its opening parenthesis.

File: base/applications/sndrec32/CMakeLists.txt (hunk 25 above)
Comment (JoachimHenze) [reply base/applications/sndrec32/CMakeLists.txt:18]: 1.) It would be more correct, As XPSP3 does not have that static import to uxtheme in sndrec32. I double-checked with MSDepends for XPSP3. 2k3sp2 with almost certainty also doesn't have that.
2.) It will continue to allow running sndrec in Win2000, XPSP3/2k3sp2-nlite-slipstreamed. And maybe also on XPSP2embedded, where it might be possible to exclude uxtheme as well.

Comment (learn-more) [reply base/applications/sndrec32/CMakeLists.txt:18]: We don't target `Win2000, XPSP3/2k3sp2-nlite-slipstreamed. And maybe also on XPSP2embedded`

Comment (JoachimHenze) [reply base/applications/sndrec32/CMakeLists.txt]: But we do target 2k3sp2. And do intend to do the things like 2k3sp2 does them.
